python/tank/deploy/tank_commands/core_upgrade.py

```diff
@@ -251,184 +256,133 @@ def __init__(self, installation_root, logger, core_version=None):
-                              ["sg_status_list", "is_not", "bad" ]]
-
-        if version_requested is not None:
-            version_filter.append(["code", "is", version_requested])
-
-        # connect to the app store
-        version_found = self._sg.find_one(constants.TANK_CORE_VERSION_ENTITY,
-                                         filters=version_filter,
-                                         fields=["sg_min_shotgun_version",
-                                                 "code",
-                                                 "sg_detailed_release_notes",
-                                                 "description",
-                                                 constants.TANK_CODE_PAYLOAD_FIELD],
-                                         order=[{"field_name": "created_at", "direction": "desc"}])
-
-        if version_found is None:
-            if version_requested:
-                raise TankError("Could not find version '%s' of the Core API in the app store!" % version_requested)
-            else:
-                # technical problems?
-                raise TankError("Could not find any version of the Core API in the app store!")
-
-        return version_found
-
     def get_update_version_number(self):
         """
         Returns the new version of the Toolkit API from shotgun
         Returns None if there is no new version
         """
-        return self._update_version["code"]
+        return self._new_core_descriptor.get_version()
 
     def get_current_version_number(self):
         """
         Returns the currently installed version of the Toolkit API
         """
-        return self._current_ver
+        return self._current_core_descriptor.get_version()
 
     def get_required_sg_version_for_update(self):
         """
         Returns the SG version that is required in order to update to the specified
         version of the Tank Core API.
-        
+
         :returns: sg version number as a string or None if no version is required. 
         """
-        return self._update_version["sg_min_shotgun_version"]
+        # FIXME: This should look inside Shotgun.
+        return self._new_core_descriptor.get_version_constraints()["min_sg"]
 
     def get_release_notes(self):
         """
         Returns the release notes for the most recent version of the Toolkit API
-        
+
         :returns: tuple with (summary_string, details_url_string)
         """
-        summary = self._update_version.get("description", "")
-        if self._update_version:
-            url = self._update_version["sg_detailed_release_notes"].get("url", "")
-        else:
-            url = ""
-        return (summary, url)
+        return self._new_core_descriptor.get_changelog()
 
     def get_update_status(self):
         """
         Returns true if an update is recommended
         """
-        if self._current_ver == "HEAD":
+        if self._current_core_descriptor.get_version() == "HEAD":
```

Comment:
I'm looking at the rest of the code that references HEAD. I'd argue that actually we should introduce a is_version_head method or something similar since we are doing case insensitive comparisons elsewhere in the core (e.g. in deploy/util.py). I've added such a method inside deploy/util